modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxnStateRocksDbStorage.java

```diff
@@ -18,243 +18,92 @@
-
-    /** RocksDB flusher instance. */
-    private volatile RocksDbFlusher flusher;
-
+public class TxnStateRocksDbStorage implements TxnStateStorage {
     /** RocksDB database. */
     private volatile TransactionDB db;
 
-    /** RockDB options. */
-    @Nullable
-    private volatile Options options;
-
-    /** Database options. */
-    private volatile TransactionDBOptions txDbOptions;
-
     /** Write options. */
-    private final WriteOptions writeOptions = new WriteOptions().setDisableWAL(true);
+    private final WriteOptions writeOptions;
 
     /** Read options for regular reads. */
-    private final ReadOptions readOptions = new ReadOptions();
-
-    /** Read options for reading persisted data. */
-    private final ReadOptions persistedTierReadOptions = new ReadOptions().setReadTier(PERSISTED_TIER);
-
-    /** Scheduled pool for {@link RocksDbFlusher}. */
-    private final ScheduledExecutorService scheduledPool;
+    private final ReadOptions readOptions;
 
-    /** Thread-pool for snapshot operations execution. */
-    private final ExecutorService threadPool;
+    /** Partition id. */
+    private final int partitionId;
 
-    /** Delay supplier for {@link RocksDbFlusher}. */
-    private final IntSupplier delaySupplier;
-
-    /** On-heap-cached last applied index value. */
-    private volatile long lastAppliedIndex;
-
-    /** The value of {@link #lastAppliedIndex} persisted to the device at this moment. */
-    private volatile long persistedIndex;
-
-    /** Whether is started. */
-    private boolean isStarted;
+    /** Transaction state table storage. */
+    private final TxnStateRocksDbTableStorage tableStorage;
 
     /** Collection of opened RocksDB iterators. */
     private final Set<RocksIterator> iterators = ConcurrentHashMap.newKeySet();
 
     /** Busy lock to stop synchronously. */
-    final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
-
-    /** Prevents double stopping the component. */
-    private final AtomicBoolean stopGuard = new AtomicBoolean();
+    private final IgniteSpinBusyLock busyLock;
 
     /**
      * The constructor.
      *
-     * @param dbPath Database path.
-     * @param scheduledPool Scheduled thread pool.
-     * @param threadPool Thread pool.
-     * @param delaySupplier Supplier of delay values to batch independent flush requests. Please refer to {@link RocksDbFlusher} for
-     *      details.
+     * @param db Database..
+     * @param busyLock Busy lock, provided by table storage.
+     * @param partitionId Partition id.
+     * @param tableStorage Table storage.
      */
-    public TxStateRocksDbStorage(
-            Path dbPath,
-            ScheduledExecutorService scheduledPool,
-            ExecutorService threadPool,
-            IntSupplier delaySupplier
+    public TxnStateRocksDbStorage(
+            TransactionDB db,
+            WriteOptions writeOptions,
+            ReadOptions readOptions,
+            IgniteSpinBusyLock busyLock,
```

Comment:
Why is this a parameter? It protects a table from stopping inconsistently, but not a partition

Reply:
in this particular case, it prevents updates for underlying table storage database